kategory-core/src/main/kotlin/kategory/typeclasses/Order.kt

```diff
@@ -0,0 +1,50 @@
+package kategory
+
+interface Order<F> : Eq<F>, Typeclass {
+
+    /**
+     * Compare [a] with [b]. Returns an Int whose sign is:
+     * - negative if `x < y`
+     * - zero     if `x = y`
+     * - positive if `x > y`
+     */
+    fun compare(a: F, b: F): Int
+
+    override fun eqv(a: F, b: F): Boolean = compare(a, b) == 0
+
+    fun lt(a: F, b: F): Boolean = compare(a, b) < 0
```

Comment:
shouldn't these functions be infix or have infix alternatives ? you should be able to compare two orderable elements by `a lt b`

Reply:
infix is only allowed on single parameter so to achieve this we could make an overload like so

```kotlin
inline fun <reified F> F.lt(O: Order<F> = order(), b: F): Boolean = O.lt(this, b)

inline infix fun <reified F> F.lt(b: F): Boolean = order<F>().lt(this, b)
```

I have not introduced this because it was also not done for Eq. I did do a similar overload for a different reason here https://github.com/kategory/kategory/pull/304 which is still open with a question about the overload.

Biggest issue is that we do not have scoped implicits so there is no way to overwrite the typeclass for a certain infix (overloaded) call.

Reply:
Users can still change it globally in the worst case, not ideal but is what we have right now.